File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.py
class Solution:
    def judgeSquareSum(self, c: int) -> bool:
        sr = math.isqrt(c)
        l,r = 0, sr
        while l <= r:
            curr = l*l + r*r
            if curr > c:
                r-=1
            elif curr < c:
                l+=1
            else:
                return True
        return False

File: 0633-sum-of-square-numbers/0633-sum-of-square-numbers.go
func judgeSquareSum(c int) bool {
    sr := int(math.Sqrt(float64(c)))
    l, r := 0, sr

    for l <= r{
        curr := l*l + r*r
        if curr > c {
            r-- 
        }else if curr < c {
            l++
        }else {
            return true
        }
    }
    return false
}

